### PR TITLE
Add additional permission for pull payments

### PR DESCRIPTION
**Pull request by Kukks**
cc @ndeet 
There are 2 new permissions `CanCreatePullPayments` and `CanCreateNonApprovedPullPayments`.

I've also refactored the way to register sub-policies as it was getting very hard to follow the tree

### CODE REVIEW

**Comment by ndeet** [comment]
Awesome, thanks a lot 💚 will try and let you know.

**Comment by ndeet** [comment]
Can confirm the permission works! Thank you. 

The description of the new permissions is the same as on the primary one, needs to be adjusted: 
image

On the payouts, not sure how it looked before or if it was the same. On the pull payment page that orange indicator had a text "awaiting approval" not sure if it should lile this on the payouts page:

image


Running regtest on your branch which seems to be on 1.7.4.0

File: BTCPayServer/Controllers/UIManageController.APIKeys.cs [hunk 1]
```diff
@@ -540,6 +540,10 @@ public class PermissionValueItem
                     {$"{Policies.CanViewPaymentRequests}:", ("View your payment requests", "The app will be able to view the selected stores' payment requests.")},
                     {Policies.CanManagePullPayments, ("Manage your pull payments", "The app will be able to view, modify, delete and create pull payments on all your stores.")},
                     {$"{Policies.CanManagePullPayments}:", ("Manage selected stores' pull payments", "The app will be able to view, modify, delete and create new pull payments on the selected stores.")},
+                    {Policies.CanCreatePullPayments, ("Create pull payments", "The app will be able to view, modify, delete and create pull payments on all your stores.")},
```
Comment (Kukks) [inline BTCPayServer/Controllers/UIManageController.APIKeys.cs:543]:
```suggestion
                    {Policies.CanCreatePullPayments, ("Create pull payments", "The app will be able to create pull payments on all your stores.")},
```

File: BTCPayServer/Controllers/UIManageController.APIKeys.cs [hunk 2]
```diff
@@ -540,6 +540,10 @@ public class PermissionValueItem
...
+                    {$"{Policies.CanCreatePullPayments}:", ("Create pull payments in selected stores", "The app will be able to view, modify, delete and create new pull payments on the selected stores.")},
```
Comment (Kukks) [inline BTCPayServer/Controllers/UIManageController.APIKeys.cs:544]:
```suggestion
                    {$"{Policies.CanCreatePullPayments}:", ("Create pull payments in selected stores", "The app will be able to create new pull payments on the selected stores.")},
```

File: BTCPayServer/Controllers/UIManageController.APIKeys.cs [hunk 3]
```diff
@@ -540,6 +540,10 @@ public class PermissionValueItem
...
+                    {Policies.CanCreateNonApprovedPullPayments, ("Create non-approved pull payments", "The app will be able to view, modify, delete and create pull payments on all your stores.")},
```
Comment (Kukks) [inline BTCPayServer/Controllers/UIManageController.APIKeys.cs:545]:
```suggestion
                    {Policies.CanCreateNonApprovedPullPayments, ("Create non-approved pull payments", "The app will be able to create pull payments without automatic approval on all your stores.")},
```

File: BTCPayServer/Controllers/UIManageController.APIKeys.cs [hunk 4]
```diff
@@ -540,6 +540,10 @@ public class PermissionValueItem
...
+                    {$"{Policies.CanCreateNonApprovedPullPayments}:", ("Create non-approved pull payments in selected stores", "The app will be able to view, modify, delete and create new pull payments on the selected stores.")},
```
Comment (Kukks) [inline BTCPayServer/Controllers/UIManageController.APIKeys.cs:546]:
```suggestion
                    {$"{Policies.CanCreateNonApprovedPullPayments}:", ("Create non-approved pull payments in selected stores", "The app will be able to view, modify, delete and create pull payments without automatic approval on the selected stores.")},
```

**Review by dennisreimann — APPROVED** [review]

**Comment by Kukks** [comment]
> Can confirm the permission works! Thank you. 
> 
> The description of the new permissions is the same as on the primary one, needs to be adjusted: 
> image
> 
> On the payouts, not sure how it looked before or if it was the same. On the pull payment page that orange indicator had a text "awaiting approval" not sure if it should lile this on the payouts page:
> 
> image
> 
> 
> Running regtest on your branch which seems to be on 1.7.4.0 

Done, although the progress bar has been like that and is unrelated to this PR. Will let nicolas review first since I refactored some sensitive subpolicy

**Comment by NicolasDorier** [comment]
The orange bar seems correct to me? the refund hasn't happen yet, so shouldn't be green.
I checked that the permission tree is the same and that's fine.

**Comment by ndeet** [comment]
re orange bar, yes seems it was always like that, I think I was confused as on the screen before there was also a bar but with some text in it and I thought there might be something missing
